docs: update s3 doc to indicate endpoint shouldn't include the bucket name prefix (#11039)
grafana/mimir: docs/sources/mimir/configure/configure-object-storage-backend.md

    bucket_name: mimir-ruler
```

{{< admonition type="note" >}}

> The endpoint value must not be prefixed with your bucket name (e.g., if your bucket is "my-storage-bucket", the endpoint cannot be "my-storage-bucket.s3.amazonaws.com").

Reviewer: ```suggestion
Don't prefix the endpoint value with your bucket name. For example, if your bucket is named "my-storage-bucket", you cannot name the endpoint "my-storage-bucket.s3.amazonaws.com").
```

Author: @tacole02 I've addressed the comment, please have a look
